tools/data_builder.py

```diff
@@ -119,6 +122,8 @@ def parse_state(state, text):
 
     if entry and entry["links"]:
         yield entry
+    else:
+        print(f"Failed links parse: missing links for {entry}")

```

Comment:
This is not a failure mode, I think it's the norm. Perhaps a comment is missing there, that it's trying to make sure if we didn't find another city or title, then to submit the last entry we had.

The thing to print in that `else` case is "Finished processing this state" if you really wanted to print something

Reply:
If you run this against the existing data set, this will only print on entries that have no links (e.g. Kansas.md's `(no entries yet)`) - the `yield` on 124 above is actually the 'finished' state

Reply:
Then I misunderstood. Thank you.